packages/opentelemetry-context-zone-peer-dep/src/ZoneContextManager.ts

```diff
@@ -205,7 +205,7 @@ export class ZoneContextManager implements ContextManager {
    * @param target
    * @param context A context (span) to be bind to target
    */
-  bind<T>(target: T | TargetWithEvents, context: Context): T {
+  bind<T>(context: Context, target: T | TargetWithEvents): T {
```

Comment:
Just a logic thing (and probably something I missed somewhere), but generally I'd expect an argument called "target" to be the "this" (for a bind) rather than the first argument -- but as the first argument "target" makes sense...

Perhaps this potential confusion can be rectified by adding some documentation to the tsdoc param.

Reply:
I think you are conflating `target` and `this`, but I'll fix the tsdoc

Reply:
Updated the tsdocs. Let me know if its more clear.

Reply:
Correct,

That's what I'm calling out i.e. the "special" case nature of what bind() is doing vs ```myClass.action(target)```, even just renaming the ```context``` to point out the ```this``` nature would probably be sufficient.
Especially, as this is the classes version of bind and not Function.prototype.bind (which calls this first argument ```thisArg```)

So perhaps calling it ```contextThis``` (or something similar) for example would be useful.

Reply:
Bind doesn't affect `this` and has nothing to do with it. I'm not sure what you're suggesting here.

Reply:
Ok, total confusion on my part around the overloaded usage of ```bind``` here vs using the JS ```bind``` (Function.prototype.bind) to set the this...

Especially as the bind here is really (setting) by calling _bindListener() [with the arguments ordered as id expect].

Way too many overloaded terms in the spec between languages -- especially when jumping between projects *sigh* -- lets call ```this``` a newbie issue (can I still claim that 😁 )